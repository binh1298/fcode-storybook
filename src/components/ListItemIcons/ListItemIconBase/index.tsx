import {
    ListItemIcon as MaterialListItemIcon,
    ListItemIconProps as MaterialListItemIconProps,
} from "@material-ui/core";

export interface ListItemIconBaseProps extends MaterialListItemIconProps {}

const ListItemIconBase = (props: ListItemIconBaseProps) => {
    return <MaterialListItemIcon {...props} />;
};

export default ListItemIconBase;
