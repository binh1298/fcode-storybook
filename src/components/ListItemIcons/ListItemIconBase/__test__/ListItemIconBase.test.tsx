import ListItemIconBase, { ListItemIconBaseProps } from "../";

import { render, RenderResult, screen } from "@testing-library/react";

describe("<ListItemIconBase />", () => {
    let wrapper: RenderResult;

    const props: ListItemIconBaseProps = {
        children: <div>Test</div>,
    };

    beforeEach(() => {
        wrapper = render(<ListItemIconBase {...props} />);
    });

    it("should exist children", () => {
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});
