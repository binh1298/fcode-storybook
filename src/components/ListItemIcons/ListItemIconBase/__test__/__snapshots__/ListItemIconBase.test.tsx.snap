// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemIconBase /> should match snapshot 1`] = `
<div>
  <div
    class="MuiListItemIcon-root"
  >
    <div>
      Test
    </div>
  </div>
</div>
`;
